File: test/PUB-SUB/SUB-unikernel/non-unikernel/pub.py
import zmq
import time
context = zmq.Context()

#  Socket to talk to server
print("Starting server...")
socket = context.socket(zmq.PUB)
socket.bind("tcp://127.0.0.1:5556")
while True:
    socket.send(b"ABC")
    #print(socket.recv_string())

File: perf/PUB-SUB/java_server/src/main/java/Pub.java
import org.zeromq.ZMQ;
import org.zeromq.ZContext;

public class Pub
{
    public static void main(String[] args) throws Exception
    {
        try (ZContext context = new ZContext()) {
            ZMQ.Socket socket = context.createSocket(ZMQ.PUB);
            socket.bind("tcp://*:5555");

            while (!Thread.currentThread().isInterrupted()) {
                String content = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
                socket.send(content.getBytes(ZMQ.CHARSET), 0);
            }
        }
    }
}
